### PR TITLE
feat: add OSS configurations

### PR DESCRIPTION
**Pull request by michelkaporin**
- Adds ability to enable/disable OSS feature
- Adds ability to filter results in UI for all features
- Adds ability to pass additional arguments to CLI 

Screenshot 2021-09-21 at 17 21 12

Screenshot 2021-09-21 at 17 23 20

**Screenshot with PR review changes incorporated:**
Screenshot 2021-09-30 at 08 26 29

### CODE REVIEW

**Comment by ArtsiomCh** [comment]
With Severity filter usage. OSS result's Icons shown correctly, while Snyk Code always tend to show Low icon if nothing to show:
image
image

**Comment by ArtsiomCh** [comment]
For `npm` projects doing OSS re-scan without previous `npm install` has no sense:
image
Shell we do `npm install` before each re-scan in such cases and if so, how that may affect performance and UX?

**Comment by ArtsiomCh** [comment]
Internal naming `Oss` seems to be a bit confusing in UI:
image
image
Shell we use full name instead?

File: src/snyk/cli/services/cliService.ts [hunk 1]
```diff
@@ -101,6 +103,11 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
     args.push(...foldersToTest);
     args.push('--json');
 
+    const additionalParams = this.config.getAdditionalCliParameters();
+    if (additionalParams) {
+      args.push(...additionalParams.trim().split(' '));
```
Comment (ArtsiomCh) [inline src/snyk/cli/services/cliService.ts:108]: if params holds space inside (for example valid case with `--json-file-output="g o o f.json"` ) we will fail

File: src/snyk/snykCode/views/issueTreeProvider.ts [hunk 2]
```diff
@@ -150,4 +154,20 @@ export class IssueTreeProvider extends AnalysisTreeNodeProvder {
...
+      switch (issue.severity) {
+        case DiagnosticSeverity.Error:
+          return this.configuration.severityFilter.high;
+        case DiagnosticSeverity.Warning:
+          return this.configuration.severityFilter.medium;
+        case DiagnosticSeverity.Information:
+        case DiagnosticSeverity.Hint:
+          return this.configuration.severityFilter.low;
+        default:
+          return true;
```
Comment (ArtsiomCh) [inline src/snyk/snykCode/views/issueTreeProvider.ts:169]: should we prepare for `critical` level issues? afaik it will be introducing in Snyk code soon.

Comment (michelkaporin) [reply src/snyk/snykCode/views/issueTreeProvider.ts:169]: Let's address it when it comes. It is not a scope of this PR.

Comment (michelkaporin) [reply src/snyk/snykCode/views/issueTreeProvider.ts:169]: (There are more changes than just this place to be edited for that).

**Comment by michelkaporin** [comment]
> With Severity filter usage. OSS result's Icons shown correctly, while Snyk Code always tend to show Low icon if nothing to show:
@ArtsiomCh It's a great catch! I've spoken to @andygongea and he also suggested to add a note when no severity filters are selected (possible due to nature of settings in VS Code) to prevent situations when users won't understand why they don't see any results.

**Comment by michelkaporin** [comment]
> For npm projects doing OSS re-scan without previous npm install has no sense:
image
Shell we do npm install before each re-scan in such cases and if so, how that may affect performance and UX?

It makes sense, e.g. if dependency is removed totally from package.json or both package-lock.json and package.json is updated.

This is however a very good question that raises again a question how valuable auto scan is at all, cc @deepcodeg. We cannot run `npm install` or other command without requesting user for such action. The best solution in this case to me is to remove auto-scan at all when we introduce remediation through `snyk fix`. CLI will do the job for us by upgrading the required dependency.

**Comment by michelkaporin** [comment]
>Internal naming Oss seems to be a bit confusing in UI:
Shell we use full name instead?

Valid concern, I was cautious about settings length, however it's always better to be specific rather than leaving space for ambiguity. Updated now :)

**Review by ArtsiomCh — APPROVED** [review]
LGTM

**Pull request merged**